### PR TITLE
ci: update cilium-cli to v0.10.4

### PR DESCRIPTION
**Pull request by tklauser**
See individual commits.

### CODE REVIEW

**Comment by tklauser** [comment]
Affected CI workflows passed, removing test commit and marking as ready for review.

File: .github/workflows/conformance-externalworkloads-v1.10.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ env:
   vmName: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}-vm
   zone: us-west2-a
   vmStartupScript: .github/gcp-vm-startup.sh
-  cilium_cli_version: v0.10.3
+  cilium_cli_version: v0.10.4
```
Comment (pchaigno) [inline .github/workflows/conformance-externalworkloads-v1.10.yml:68]: Don't we need to set the base version here as well?

File: .github/workflows/conformance-externalworkloads-v1.11.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ env:
   vmName: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}-vm
   zone: us-west2-a
   vmStartupScript: .github/gcp-vm-startup.sh
-  cilium_cli_version: v0.10.3
+  cilium_cli_version: v0.10.4
```
Comment (pchaigno) [inline .github/workflows/conformance-externalworkloads-v1.11.yml:68]: Same here.

File: .github/workflows/conformance-multicluster-v1.10.yaml [hunk 3]
```diff
@@ -147,9 +147,12 @@ jobs:
             --relay-version=${SHA}"
           CLUSTERMESH_ENABLE_DEFAULTS="--apiserver-image=quay.io/${{ github.repository_owner }}/clustermesh-apiserver-ci \
             --apiserver-version=${SHA}"
+          CONNECTIVITY_TEST_DEFAULTS="--base-version=v1.10 \
+            --flow-validation=disabled"
```
Comment (pchaigno) [inline .github/workflows/conformance-multicluster-v1.10.yaml]: Why are we disabling flow validation here?

File: .github/workflows/conformance-externalworkloads-v1.10.yml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:68]: Good catch, I missed this one. Changed to set base version v1.10.

File: .github/workflows/conformance-externalworkloads-v1.11.yml (hunk 2 above)
Comment (tklauser) [reply .github/workflows/conformance-externalworkloads-v1.11.yml:68]: Changed to set base version v1.11.

File: .github/workflows/conformance-multicluster-v1.10.yaml (hunk 3 above)
Comment (tklauser) [reply .github/workflows/conformance-multicluster-v1.10.yaml]: Mindless copy&paste, it seems. Reverted that change.

**Comment by tklauser** [comment]
CI passed on all affected workflows, removing test commit again.

**Review by pchaigno — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Pull request merged**